study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/RarexVersion2.java

```diff
@@ -0,0 +1,723 @@
+        questionDao.insertQuestion(generalInformationId, questionDef, questionDto.get().getRevisionId());
+        questionDao.getJdbiBlockQuestion().insert(blockId, questionDef.getQuestionId());
+
+        LOG.info("REFID 30");
+        blockId = helper.findBlockByText(healthAndDevelopmentId,
+                "While your mom was pregnant with you do you know if there were any concerns with the pregnancy or "
+                        + "have you been told that there were issues during <u>your biological mother's</u> PREGNANCY with you?*");
+        helper.updateBlockVariable(blockId, "question", "Have you been told that there were issues during <u>your biological "
+                + "mother's</u> PREGNANCY with you and/or your DELIVERY/BIRTH?*");
+        helper.updateBlockVariable(blockId, "explanation", "Examples: Mother had to be put on bed rest, "
+                + "there were concerns on ultrasound, you were born early (prematurely), or you had to stay in the neonatal "
+                + "intensive care unit (NICU).");
+        blockId = helper.findBlockByText(healthAndDevelopmentId,
+                "During the pregnancy with this patient do you know if there were any concerns with the pregnancy or "
+                        + "have you been told that there were issues during the biological mother's pregnancy?*");
+        helper.updateBlockVariable(blockId, "question", "Have you been told that there were issues during <u>"
+                + "the patient's biological mother's</u> PREGNANCY with the patient and/or the patient DELIVERY/BIRTH?*");
+        helper.updateBlockVariable(blockId, "explanation", "Examples: Mother had to be put on bed rest, "
+                + "there were concerns on ultrasound, the patient was born early (prematurely), or the patient had to stay "
+                + "in the neonatal intensive care unit (NICU).");
+        blockId = helper.findBlockByText(healthAndDevelopmentId, "Examples: You were born early (prematurely), or you "
+                + "had to stay in the neonatal intensive care unit (NICU).");
+        helper._deleteSectionBlockMembershipByBlockId(blockId);
+        blockId = helper.findBlockByText(healthAndDevelopmentId, "Examples: The patient was born early (prematurely), or "
+                + "had to stay in the neonatal intensive care unit (NICU).");
+        helper._deleteSectionBlockMembershipByBlockId(blockId);
+        detachQuestionFromBothSectionAndBlock(handle, "HD_LABOR_AND_DELIVERY_ISSUES");
+    }
+
+    private void updateExpressionTextByQuestionStableCode(long activityId, String questionStableCode, String expr) {
+        List<Long> ids = helper.findBlockExpressionIdsByQuestionStableCode(activityId, questionStableCode);
+        helper.updateExpressionText(expr, ids);
+    }
+
+    private void updateExpressionText(long activityId, String before, String after) {
+        List<Long> ids = helper.findBlockExpressionIdsByExpressionText(activityId, before);
```

Comment:
Can you do checks on the row counts coming back from all these queries to ensure that the right number of things are being updated?  It's a good sanity check that can catch query errors that might update unrelated studies/forms.

Reply:
That's why it's narrowed down by activityId and also logged. But still it's a good idea, I'll do this.